crates/rpc/rpc/src/otterscan.rs

```diff
@@ -198,10 +198,61 @@ where
     /// Handler for `getTransactionBySenderAndNonce`
     async fn get_transaction_by_sender_and_nonce(
         &self,
-        _sender: Address,
-        _nonce: u64,
-    ) -> RpcResult<Option<Transaction>> {
-        Err(internal_rpc_err("unimplemented"))
+        sender: Address,
+        nonce: u64,
+    ) -> RpcResult<Option<TxHash>> {
+        // Check if the sender is a contract
+        if self.has_code(sender, None).await? {
+            return Ok(None)
+        }
+
+        let highest =
+            EthApiServer::transaction_count(&self.eth, sender, None).await?.saturating_to::<u64>();
+
+        // If the nonce is higher than the highest nonce, the transaction is pending or not exists
+        if nonce >= highest {
```

Comment:
```suggestion
        if nonce > highest {
```

Reply:
or is it code comment that needs update?

Reply:
thanks, update the comments